### DISCUSSION TITLE
Is Insider the only way to get different social cards / blog post?

### DISCUSSION THREAD

**Discussion by johnsturgeon** [squidfunk/mkdocs-material#7331] [Q&A] [answered]
I'm interested in customizing the icon for different blog posts, and I was wondering if there was a way to accomplish this without Insider.

**Comment by kamilkrzyskow** [comment, marked as answer]
Hello @johnsturgeon,
it depends on what are you after specifically. If you're asking about configuring the social cards layouts on the community version, then it's impossible, as the layouts were introduced in the Insiders version, so there is no easy straight-forward configuration for this.
https://github.com/squidfunk/mkdocs-material/blob/36e5263c00fdaf147e37ad4aa65f0277e06d9fdc/material/plugins/social/config.py#L28-L36

However, since both MkDocs and the plugin are written in Python, everything is public, and you can monkey-patch functions with a [hook](https://www.mkdocs.org/user-guide/configuration/#hooks) or inside the CI before deployment you could inject other code into the plugin script file to change the behaviour.

Only caveat of this is that the fancy multi-layer code was also introduced in Insiders, so depending on how much control you want over the positioning of the elements etc. you'd have to add a lot of the code. If you only want to change the logo on a certain social card then I guess ~~monkey-patching the `self._resized_logo_promise.result()` call between pages~~ or redefining the `_render_card` function would suffice.

EDIT: The first proposed idea will likely not work as the `_render_card` function runs concurrently, so depending on the race condition the wrong logo could be used on a given social card. So `_render_card` redefinition is likely the better way. 

EDIT2:
Apart of that you could patch the `on_page_markdown` directly, but be aware that you can't just patch it the "normal" way over the class reference, as during config load, each event function is registered inside MkDocs, so you have to modify that reference.

https://github.com/squidfunk/mkdocs-material/blob/36e5263c00fdaf147e37ad4aa65f0277e06d9fdc/material/plugins/social/plugin.py#L250-L281

**Reply by johnsturgeon** [reply]
Thanks for the info!
